File: c0-analysis/cslibs/conio.cs

public class _l_conio
{
	public static void print(string s)
	{
		System.Console.Write(s);
	}

	public static void println (string s)
	{
		System.Console.WriteLine(s);
	}

	public static void printint(int n)
	{
		System.Console.Write(n);
	}

	public static void printbool(bool b)
	{
		System.Console.Write(b);
	}

	public static void printchar(char c)
	{
		System.Console.Write(c);
	}

	public static string readline()
	{
		return System.Console.ReadLine();
	}

	public static void error(string s)
	{
		C0Program.assert(false, s);
	}
}

File: c0-analysis/libs/conio/conio.c
#include <stdio.h>
#include <c0runtime.h>
#include "util.h"

void print(c0_string s) {
  const char *cstr = c0_string_tocstr(s);
  printf("%s", cstr);
  c0_string_freecstr(cstr);
}

void println(c0_string s) {
  const char *cstr = c0_string_tocstr(s);
  puts(cstr);
  c0_string_freecstr(cstr);
}

void printint(int i) {
  printf("%d", i);
}

void printbool(bool b) {
  puts(b ? "true" : "false");
}

void printchar(char c) {
  putchar(c);
}

c0_string readline() {
  return freadline(stdin);
}

void error(c0_string s) {
  const char *cstr = c0_string_tocstr(s);
  c0_abort(cstr);
  /* should be unnecessary, since c0_abort will exit: */
  c0_string_freecstr(cstr);
}